Argo/Globals/Parser.swift

```diff
@@ -0,0 +1,29 @@
+public enum Parser<T> {
+  case Success(Box<T>)
+  case TypeMismatch(String)
+  case MissingKey(String)
+
+  public var value: T? {
+    switch self {
+    case let .Success(box): return box.value
+    default: return .None
+    }
+  }
+}
+
+public func typeMismatch<T>(type: String, object: JSON) -> Parser<T> {
```

Comment:
So it looks like we _would_ need to expose this to users in order to let them write custom parsers. In that case, can we move these onto `Parser` as static functions? We can rely on type inference to let users write `.typeMismatch` instead of `Parser.typeMismatch`, and we won't pollute the global scope.